api/v1alpha3/awscluster_webhook.go

```diff
@@ -96,6 +97,14 @@ func (r *AWSCluster) ValidateUpdate(old runtime.Object) error {
 		)
 	}
 
+	// If a principalRef is already set, do not allow removal of it.
+	if oldC.Spec.PrincipalRef != nil && r.Spec.PrincipalRef == nil {
+		allErrs = append(allErrs,
```

Comment:
Do we want to allow the changing of the principleref at all? Whats the worst that can happen if they change it so the creds are for a different account? The old cluster would be orphaned and a new one created.

Reply:
We want to allow the change to support this case (from the proposal):

```
The PrincipalRef field will be mutable in order to support clusterctl move scenarios where a user instantiates a cluster on their laptop and then makes the cluster self-managed.


```

Reply:
Can you change the namespace as part of the move?

Reply:
IIRC that is not possible.

Reply:
Principals either need to be in the same namespace as the resource or be a cluster scoped resource.

Reply:
See my comment above about not using Object Reference.